gobblin-data-management/src/main/java/gobblin/data/management/copy/entities/CommitStepCopyEntity.java

```diff
@@ -27,20 +28,25 @@
 @EqualsAndHashCode(callSuper = true)
 public class CommitStepCopyEntity extends CopyEntity {
 
-  @Getter
-  private final CommitStep step;
   /** A priority value that can be used for sorting {@link CommitStepCopyEntity}s. Lower values are higher priority.*/
   @Getter
   private final int priority;
+  private final String key;
 
-  public CommitStepCopyEntity(String fileSet, Map<String, Object> additionalMetadata, CommitStep step, int priority) {
+  public CommitStepCopyEntity(String fileSet, Map<String, Object> additionalMetadata, CommitStep step, int priority,
+      CommitStepDB commitStepDB, String stepKey) throws IOException {
```

Comment:
Do you actually need to pass around ```CommitStepDB`` everywhere? Since all of them connect to the same db and use the same table I think having a singleton class is better.